src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchOperations.java

```diff
@@ -406,4 +407,26 @@ default String stringIdRepresentation(@Nullable Object id) {
 		return Objects.toString(id, null);
 	}
 	// endregion
+
+	/**
+	 * Execute a callback with the {@link RestHighLevelClient}
+	 * 
+	 * @param callback the callback to execute, must not be {@literal null}
+	 * @param <T> the type returned from the callback
+	 * @return the callback result
+	 * @throws UnsupportedOperationException for the TransportClient based implementation.
+	 * @since 4.0
+	 */
+	<T> T execute(RestHighLevelClientCallback<T> callback) throws RuntimeException;
```

Comment:
Let's expose this method on the REST Template only for a more concise applicability. Teasing client callback and then explaining that it does not work for the Transport Client is confusing.

The thrown type should be `DataAccessException`.